src/words/environment/WordsObject.java

```diff
@@ -84,8 +108,16 @@ public void setProperty(String propertyName, Property property) throws WordsRunt
 
 		if (properties.containsKey(propertyName) && property.type == Property.PropertyType.NOTHING)
 			properties.remove(propertyName);
```

Comment:
I see, that's pretty cool.  Just to make sure I get it, object B's "referers" is not really the *current* set of object properties pointing to B but the set of *all* object properties that ever at one time had a property pointing to B.  Some of those may no longer be attached to their original object (A), but that's OK, we'll set them to NOTHING anyway.  And then they'll get garbage collected by Java shortly after, because they get removed from "referers" so no one has a handle to them any more.